audio-echo/app/build.gradle

```diff
@@ -6,22 +6,18 @@ android {
     defaultConfig {
         applicationId 'com.google.sample.echo'
         /*
-           To run on earlier version of Android than android-21, do:
+           To run on earlier version of Android other than android-23, do:
              *) set this minSDKVersion and cmake's ANDROID_PLATFORM to your version
-             *) set ANDROID_STL to c++_static for some very earlier version android.
+             *) set ANDROID_STL to c++_static for earlier Android API levels
          */
         minSdkVersion 23
         targetSdkVersion 25
         versionCode 1
         versionName '1.0'
-        ndk {
-            abiFilters 'armeabi-v7a', 'arm64-v8a', 'x86', 'x86_64'
-        }
+
         externalNativeBuild {
             cmake {
-                // With android studio 2.2.0 +cmake, app needs to pack shared lib to
-                // into APK. Refer to CMakeLists.txt for details
-                arguments '-DANDROID_STL=c++_shared', '-DANDROID_TOOLCHAIN=clang',
+                arguments '-DANDROID_STL=c++_shared',
                           '-DANDROID_PLATFORM=android-23'
```

Comment:
Isn't this superfluous?

Reply:
default it is 23, same as minSDKVersion. make it explicit would help for the usage; in some version of NDK, if set to a platform that does not have a android-"version" directory under platforms, it will not build; however minSDK could be set to that version (it exists).

Reply:
IMO this just clutters up the sample. The fact that this is done in all of the samples (or at least so many of them) makes it look as though the developer must do this rather than rely on `minSdkVersion`. Frankly, there are no for the developer to set this explicitly. Setting this higher than `minSdkVersion` is just broken, and setting it lower needlessly excludes APIs they might want to use.

It would make more sense to me to have *one* sample for all the different configuration options, but keep all the others minimal to make them easier to understand for a newcomer.